### PR TITLE
DAOS-10276 test: DAOS interoperability test different version of DAOS server and agent CI automation

### PR DESCRIPTION
**Pull request by dinghwah**
Description:
  DAOS-10287: Negative test mix server ranks with of 2.0 and 2.2.
  DAOS-10276: Different version of server and client.
  DAOS-10278: Verify new pool create after server upgraded to new version.
  DAOS-10283: Verify formatted 2.0 upgraded to 2.2 pool create system downgrade to 2.0
  DAOS-11689: Interoperability between different version of daos_server and daos_agent
  DAOS-11691: Interoperability between different version of daos_server and libdaos

Test step:
  (1)Setup
  (2)dmg system stop
  (3)Upgrade 1 server-host to new version
  (4)Negative test - dmg pool query on mix-version servers
  (5)Upgrade rest server-hosts to 2.2
  (6)Restart 2.0 agent
  (7)Verify 2.0 agent connect to 2.2 server, daos and libdaos
  (8)Upgrade agent to 2.2
  (9)Verify pool and containers create on 2.2 agent and server
  (10)Downgrade server to 2.0
  (11)Verify 2.2 agent to 2.0 server, daos and libdaos
  (12)Downgrade agent to 2.0

Test-tag: interop
Required-githooks: true

### CODE REVIEW

**Comment by daltonbohning** [comment]
Is this ready for review?

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/interoperability/updown_grade.py [hunk 1]
```diff
@@ -9,24 +9,25 @@
...
 
-from general_utils import get_random_bytes, pcmd, run_pcmd
 from pydaos.raw import DaosApiError
+from general_utils import get_random_bytes, pcmd, run_pcmd
 from agent_utils import include_local_host
 from command_utils_base import CommandFailure
 from ior_test_base import IorTestBase
 
-# pylint: disable=global-variable-not-assigned,global-statement
+# pyint: disable=global-variable-not-assigned,global-statement
```
Comment (daltonbohning) [inline src/tests/ftest/interoperability/updown_grade.py:18]: Typo? `pyint -> pylint`

File: src/tests/ftest/interoperability/updown_grade.py [hunk 2]
```diff
@@ -125,7 +123,78 @@ def show_daos_version(self, all_hosts, hosts_client):
...
+        cmds.append("rpm -qa | grep daos")
+        cmds_client = cmds + ["sudo cp /etc/daos/daos_agent.yml.rpmsave /etc/daos/daos_agent.yml"]
+        cmds_client += ["sudo cp /etc/daos/daos_control.yml.rpmsave /etc/daos/daos_control.yml"]
+        cmds_svr = cmds + ["sudo cp /etc/daos/daos_server.yml.rpmsave /etc/daos/daos_server.yml"]
+
+        if servers:
+            self.log.info("==upgrade_downgrading on servers: %s", servers)
+            for cmd in cmds_svr:
+                self.log.info("==cmd= %s", cmd)
+                result = run_pcmd(servers, cmd)
```
Comment (daltonbohning) [inline src/tests/ftest/interoperability/updown_grade.py:152]: Shouldn't we check the result of run_pcmd?

File: src/tests/ftest/interoperability/updown_grade.yaml [hunk 3]
```diff
@@ -14,31 +18,32 @@ server_config:
...
 pool:
   control_method: dmg
-  scm_size: 1G
+  scm_size: 5G
   name: daos_server
 container:
   control_method: daos
   type: POSIX
-  properties: rd_fac:1
+  properties: rf:1
```
Comment (daltonbohning) [inline src/tests/ftest/interoperability/updown_grade.yaml:26]: Should not be changed
```suggestion
  properties: rd_fac:1
```

**Review by daltonbohning — CHANGES_REQUESTED** [review]
There is a conflict with latest master

File: src/tests/ftest/interoperability/updown_grade.yaml [hunk 4]
```diff
@@ -1,7 +1,11 @@
 hosts:
-  test_servers: 9
-  test_clients: 1
-timeout: 850
+  test_servers:
+    - server-A
+    - server-B
+    - server-C
+  test_clients:
+    - client-A
```
[the remark this answers is not in the document]
Comment (dinghwah) [reply src/tests/ftest/interoperability/updown_grade.yaml:7]: Yes. 
However, the base RPMs the script is running still with old format.
I will update this once the base RPMs with the new format.

File: src/tests/ftest/interoperability/updown_grade.py (hunk 1 above)
Comment (dinghwah) [reply src/tests/ftest/interoperability/updown_grade.py:18]: Thanks for the comment, please see the updates.

**Comment by dinghwah** [comment]
Close this PR due to large # of files conflict.
Will reopen a new PR to commit the updown_grade.py and yaml files.

**Pull request closed without merging**